### PR TITLE
Fix building on modern Linux system

### PR DESCRIPTION
**Pull request by cryptomilk**
# Description

This does some cmake cleanup and fixes building SVT-AV1 on openSUSE Tumbleweed.

# Issue

 It fails with linking because of missing symbols for phtread and libm. The patches address those issues.

Example:

```
[   15s] /home/abuild/rpmbuild/BUILD/SVT-AV1-0.8.5/Source/Lib/Encoder/Codec/EbEntropyCoding.c:3283: undefined reference to `log2'
[   15s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /tmp/libSvtAv1Enc.so.0.8.5.egXjb6.ltrans48.ltrans.o:/home/abuild/rpmbuild/BUILD/SVT-AV1-0.8.5/Source/Lib/Encoder/Codec/EbEntropyCoding.c:3284: undefined reference to `log2'
[   15s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /tmp/libSvtAv1Enc.so.0.8.5.egXjb6.ltrans50.ltrans.o: in function `eb_shutdown_process.isra.0':
[   15s] /home/abuild/rpmbuild/BUILD/SVT-AV1-0.8.5/Source/Lib/Common/Codec/EbThreads.c:231: undefined reference to `sem_post'
```

# Author(s)
Andreas Schneider

# Performance impact
None, build system changes only

# Merge method
- [ ] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [x] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

File: Source/Lib/Decoder/CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +18,11 @@ set(DEC_VERSION ${DEC_VERSION_MAJOR}.${DEC_VERSION_MINOR}.${DEC_VERSION_PATCH})
 
 if(UNIX)
     if(NOT APPLE)
-        find_library(M_LIB name m)
+        find_library(M_LIB NAMES m)
         if(M_LIB)
             list(APPEND PLATFORM_LIBS m)
         endif()
+        list(APPEND PLATFORM_LIBS ${CMAKE_THREAD_LIBS_INIT})
```
Comment (1480c1) [inline Source/Lib/Decoder/CMakeLists.txt:25]: can you try with `Threads::Threads` instead?

Comment (cryptomilk) [reply Source/Lib/Decoder/CMakeLists.txt]: Fixed. Thanks!

**Comment by hassount** [comment]
@cryptomilk thank you for the contribution, can you please confirm that you have signed the appropriate CLA as described in this section? https://github.com/AOMediaCodec/SVT-AV1#contributor-agreement

**Review by 1480c1 — APPROVED** [review]
I like the use of `check_c_source_compiles` instead of my very simple conftest.c and also the better variable name, it would be helpful in the future if we ever decide to add support for more simds etc

**Comment by cryptomilk** [comment]
@hassount I've signed it.

@1480c1 I normally put all the configure checks in a ConfigureChecks.cmake and include it. See e.g. https://git.cryptomilk.org/projects/cmocka.git/tree/

**Pull request merged**